js/data/buffer.js

```diff
@@ -267,20 +280,4 @@ function align(value, alignment) {
     return value;
 }
 
-function checkNumber(input) {
-    if (!isNaN(parseFloat(input)) && isFinite(input)) {
-        return input;
-    } else {
-        throw new Error('Validation failure in buffer push method creation');
-    }
-}
-
-function checkAttributeTypeName(input) {
-    if (Object.keys(Buffer.AttributeType).indexOf(input) === -1) {
-        throw new Error('Validation failure in buffer push method creation');
-    } else {
-        return input;
-    }
-}
```

Comment:
This is fine. I see what you're getting at with the check being further away from the `eval`, but I have a hard time imagining a vulnerability vector that goes through `this.itemSize` or `attribute.type.name` but has no easier path.